server/service/src/preference/preferences/complex_pref.rs

```diff
@@ -1,15 +1,55 @@
-use serde::Deserialize;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
 
 use super::Preference;
 
-#[derive(Clone, Default, Deserialize)]
+#[derive(Clone, Default, Deserialize, Serialize)]
 pub struct ComplexPref {
     pub something_here: i32,
     pub something_else: String,
 }
 
-impl Preference<ComplexPref> for ComplexPref {
+pub struct ComplexOne;
+
+impl Preference for ComplexOne {
```

Comment:
So in theory this is about as hard as it would get for us to define a new preference?!